lms/djangoapps/oauth_dispatch/tests/test_dot_adapter.py

```diff
@@ -0,0 +1,82 @@
+"""
+Tests for Blocks Views
+"""
+
+from datetime import timedelta
+from unittest import skip
+
+import ddt
+from django.test import TestCase
+from django.utils.timezone import now
+from oauth2_provider import models
+
+from student.tests.factories import UserFactory
+
+from ..adapters import dot
+
+
+@ddt.ddt
+class TestAccessTokenView(TestCase):
+    """
+    Test class for AccessTokenView
+    """
+
+    adapter = dot.DOTAdapter()
+
+    def setUp(self):
+        super(TestAccessTokenView, self).setUp()
+        self.user = UserFactory()
+        self.public_client = self.adapter.create_public_client(user=self.user, client_id='public-client-id')
+        self.confidential_client = self.adapter.create_confidential_client(
+            user=self.user,
+            client_id='confidential-client-id'
+        )
+
+    def test_create_confidential_client(self):
+        self.assertIsInstance(self.confidential_client, models.Application)
+        self.assertEqual(self.confidential_client.client_id, 'confidential-client-id')
+        self.assertEqual(self.confidential_client.client_type, 'confidential')
+
+    def test_create_public_client(self):
```

Comment:
See current solution.  I kept the functionality around, but made it more general, so the tests are still around.